### ISSUE TITLE
Announce all OOO Users on schedule

### ISSUE THREAD

**Issue by shaunburdick** [shaunburdick/slack-ooo#5]
Announce each day which users are out of office

**Comment by andela-anandaa** [comment]
This is a good one. Perhaps in an org, there could be a #ooo channel from which the bot can do route announcements.

**Comment by shaunburdick** [comment]
What are you thinking?
Something like a list of channels and times to announce?

```
{
  announce: {
    channels: [ 'foo', 'bar', 'general' ],
    times: [ '8:00', '14:00' ]
  }
}
```

**Comment by andela-anandaa** [comment]
@shaunburdick That's even more flexible! Looks good. But the announcements can be non-obtrusive, so no _@channel-ing_, etc. And I don't if the bot could be smart enough not interrupt ongoing conversations?

Or, is it just better to have a dedicated channel for it?

**Comment by shaunburdick** [comment]
Yeah, I wouldn't @channel anything, that's the worst with bots. With the flexibility you could put it anywhere. I could put a check in to wait for 1 minute since last message before blasting it out

**Comment by andela-anandaa** [comment]
Sounds good! 👍

**Comment by andela-anandaa** [comment]
RE: [your comment here](https://github.com/shaunburdick/slack-ooo/issues/5#issuecomment-247113171), do you want a _bot-admin_ to configure this from Slack or you want it to be in a config file?

**Comment by shaunburdick** [comment]
I'm going to go with a config/env options

**Comment by shaunburdick** [comment]
If you could do me a favor and test out the announce branch, the functionality should be there.

**Comment by shaunburdick** [comment]
Seems to work for me but wanted to get other eyes on it. If everything works I will merge and create a new release.

**Comment by andela-anandaa** [comment]
@shaunburdick - On it, let me set it up, will let you know.

**Comment by andela-anandaa** [comment]
@shaunburdick Works fine, perhaps just 2 things to note:
- Let's specify in the README that you have to invite the bot in the respective channel. Bot only made an announcement in the invited channels
- The announcement message, no need to repeat _I'm out of office_ for every user, could just display the OOO-user message only

This is unrelated, do you know from which line of code this log comes from:

```
info: ** No persistent storage method specified! Data may be lost when process shuts down.
```

Just after the `.info`: `Starting Bot...`

**Comment by shaunburdick** [comment]
That's coming from the botkit library

**Comment by andela-anandaa** [comment]
Ok, cool. Will investigate on setting up a simple SQLite for simple persistence.

**Comment by shaunburdick** [comment]
re: The announcement message, no need to repeat I'm out of office for every user, could just display the OOO-user message only
What do you mean?
It should be displaying:

```
The following users are out of office:
> user1: user1's message
> user2: user2's message
.
.
.
> usern: usern's message
```

**Comment by andela-anandaa** [comment]
Exactly, that's what I mean. 👍

**Comment by andela-anandaa** [comment]
Sorry, my bad. It's actually me who told the bot that message _I'm out of office, reach to me through 07289090_, I thought I'd only added _reach to me through 07289090_

**Comment by andela-anandaa** [comment]
So, strike out number 2, does number 1 make sense?

**Comment by shaunburdick** [comment]
yeah, good idea. I just pushed some changes to the readme

**Comment by shaunburdick** [comment]
https://github.com/shaunburdick/slack-ooo/pull/14/commits/cd6a4e59734f8ee9779d59f37a9f1b2c4126caad

**Comment by andela-anandaa** [comment]
Good, seen it!

**Comment by shaunburdick** [comment]
Alrighty then, I merged the PR and created a new 2.1.0 release. Thanks for your help!
